test/lit/wat-kitchen-sink.wast

```diff
@@ -2379,6 +2379,84 @@
   drop
  )
 
+ ;; CHECK:      (func $call-ref (type $void)
+ ;; CHECK-NEXT:  (local $0 (ref null $void))
+ ;; CHECK-NEXT:  (local $1 (ref null $ret2))
+ ;; CHECK-NEXT:  (local $2 (ref null $many))
+ ;; CHECK-NEXT:  (local $scratch (i32 i32))
```

Comment:
Why were these scratch values created?

Reply:
Because the call to `many` returns a tuple, which is then accessed by multiple different `drop` instructions. Since we have an expression tree, each value can only be consumed by one expression (its parent) normally. Since tuples need to be accessible by multiple expressions, they need to be stashed in locals first. The scratch locals are the locals IRBuilder introduces to hold these tuples.